### PR TITLE
Updates Environment the response attributes for Delete, Create and Update API

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- For UPDATE, DELETE and CREATE API response only contains the environment name irespective of environment contains the attributes or not.

```
{
  "name": "chef-environment-887549700",
  "chef_type": "",
  "description": "",
  "json_class": "",
  "cookbook_versions": {
  },
  "default_attributes": "{}",
  "override_attributes": "{}"
}
```

### :chains: Related Resources
Changes requested for PR: https://github.com/chef/automate/issues/4545

### :+1: Definition of Done
- Make uniform environment response object, earlier it was returning environment name for DELETE, CREATE and UPDATE APIs.
- it will return the attributes if exists.

### :athletic_shoe: How to Build and Test the Change
- `rebuild components/infra-proxy-service`
- `rebuild components/automate-gateway`

- Create Environment API

```
curl -sSkH "api-token: $(get_admin_token)" -X POST https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/environments?pretty -d '{"name": "new-env-test", "description": "test"}'
{
  "name": "new-env-test",
  "chef_type": "environment",
  "description": "test",
  "json_class": "Chef::Environment",
  "cookbook_versions": {
  },
  "default_attributes": "{}",
  "override_attributes": "{}"
}
```

- Update environment API

```
curl -sSkH "api-token: $(get_admin_token)" -X PUT https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/environments/chef-environment-621641300?pretty -d '{"description": "test"}'
{
  "name": "chef-environment-621641300",
  "chef_type": "environment",
  "description": "test",
  "json_class": "Chef::Environment",
  "cookbook_versions": {
  },
  "default_attributes": "{}",
  "override_attributes": "{}"
}
```

- Delete Environment API

```
curl -sSkH "api-token: $(get_admin_token)" -X DELETE https://a2-dev.test/api/v0/infra/servers/chef-server-dev-test/orgs/chef-org-dev/environments/chef-environment-88754970?pretty
{
  "name": "chef-environment-887549700",
  "chef_type": "environment",
  "description": "",
  "json_class": "Chef::Environment",
  "cookbook_versions": {
  },
  "default_attributes": "{}",
  "override_attributes": "{}"
}
```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: components/infra-proxy-service/integration_test/databags_test.go [hunk 1]
```diff
@@ -71,6 +71,21 @@ func TestGetDatabagItems(t *testing.T) {
 	addDataBagItems(ctx, dataBagName, 10)
 
 	t.Run("items list with a per_page search param", func(t *testing.T) {
+		itemId := fmt.Sprintf("item-%d", time.Now().Nanosecond())
```
Comment (vsingh-msys) [inline components/infra-proxy-service/integration_test/databags_test.go:74]: This data bag item PR fixes for pipeline issue.

**Review by vivek-yadav — APPROVED** [review]

File: components/infra-proxy-service/integration_test/environments_test.go [hunk 2]
```diff
@@ -101,6 +104,188 @@ func TestGetEnvironments(t *testing.T) {
...
+		res, err := infraProxy.GetEnvironment(ctx, req)
+		assert.NoError(t, err)
+		assert.NotNil(t, res)
+		assert.Equal(t, name, res.GetName())
+		assert.Equal(t, "auto generate chef environment", res.GetDescription())
+		assert.Equal(t, creReq.CookbookVersions, res.CookbookVersions)
+	})
+}
+
+func TestCreateEnvironment(t *testing.T) {
```
Comment (kalroy) [inline components/infra-proxy-service/integration_test/environments_test.go:145]: Please add a test to cover Ovverride attributes too. I can only see defaultAttributes

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/infra-proxy-service/integration_test/databags_test.go [hunk 3]
```diff
@@ -84,6 +99,36 @@ func TestGetDatabagItems(t *testing.T) {
...
+			Data: &structpb.Struct{
+				Fields: map[string]*structpb.Value{
+					"id": &structpb.Value{Kind: &structpb.Value_StringValue{StringValue: itemId2}},
+				},
+			},
+		}
+		dbItem2, err := infraProxy.CreateDataBagItem(ctx, createReq2)
+		assert.NoError(t, err)
+		assert.NotNil(t, dbItem2)
+
```
[the remark this answers is not in the document]
Comment (vsingh-msys) [reply components/infra-proxy-service/integration_test/databags_test.go:131]: Some of the assertions we need to updates, Chef-server API is not so much reliable, have covered and improved data pulled in PR https://github.com/chef/automate/pull/4785/files#diff-525544f052c41c84645f03b65b7716279e2c0948e07ee53b635932981484b3fcR38-R49

File: components/infra-proxy-service/integration_test/environments_test.go (hunk 2 above)
Comment (vsingh-msys) [reply components/infra-proxy-service/integration_test/environments_test.go:145]: Added asset for both default & override attributes. Thanks!

```
assert.Equal(t, "{\"attribute1\":\"value\"}", res.DefaultAttributes)
assert.Equal(t, "{\"attribute1\":\"override\"}", res.OverrideAttributes)
```

**Review by kalroy — APPROVED** [review]
Thanks!!

**Pull request merged**